backends/tc/BUILD

```diff
@@ -0,0 +1,479 @@
+# tc backend for the P4 compiler.
+
+load("//:bazel/p4_library.bzl", "p4_ir")
```

Comment:
It is long mainly because (1) I kept the modules of the backend separate so that incremental builds work better, other backends have a few large build rules that create a library or the backend, and (2) there are some common dependencies for most modules (some Abseil modules, the frontend, and the P4 library). I think (1) is necessary for better build times when developing code, and I could not find a way to merge the common dependencies to alleviate (2). Maybe somebody more familiar with Bazel can take a look.

@smolkaj can you review the Bazel files?